WordPress/gutenberg
# Post fields: move `author` from `edit-site` to `fields` package

Part of https://github.com/WordPress/gutenberg/issues/61084

## What?

This PR moves the `author` field definition from `edit-site` to the `fields` package.

## Why?

In preparation for moving the whole `usePostFields` hook ([conversation](https://github.com/WordPress/gutenberg/pull/65196#discussion_r1777506405)).

## Testing Instructions

Visit the Pages page and verify that the "Author" field still works as expected (filters, visualization, edit).

<img width="1533" alt="Screenshot 2024-11-12 at 17 16 25" src="https://github.com/user-attachments/assets/427687db-efa6-4ec6-837b-c4002716e962">
